src/platform/android/AndroidChipPlatform-JNI.cpp

```diff
@@ -274,6 +277,72 @@ JNI_LOGGING_METHOD(void, setLogFilter)(JNIEnv * env, jclass clazz, jint level)
     SetLogFilter(category);
 }
 
+static void ENFORCE_FORMAT(3, 0) logRedirectCallback(const char * module, uint8_t category, const char * msg, va_list args)
+{
+    using namespace chip::Logging;
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    int priority = ANDROID_LOG_DEBUG;
+    switch (category)
+    {
+    case kLogCategory_Error:
+        priority = ANDROID_LOG_ERROR;
+        break;
+    case kLogCategory_Progress:
+        priority = ANDROID_LOG_INFO;
+        break;
+    case kLogCategory_Detail:
+        priority = ANDROID_LOG_DEBUG;
+        break;
+    default:
+        break;
+    }
+
+    jint jPriority  = static_cast<jint>(priority);
+    jstring jModule = env->NewStringUTF(module);
```

Comment:
please don't use NewStringUTF, and go with CharToStringUTF

https://github.com/project-chip/connectedhomeip/blob/3d69583fb644577269ee468e6a28c0f4d6e40612/src/lib/support/JniReferences.h#L237C16-L237C31

Reply:
I will change it to use CharToStringUTF.